core/src/test/java/io/grpc/internal/BinaryLogProviderTest.java

```diff
@@ -50,6 +119,134 @@ public void unavailableProvider() {
     assertSame(BinaryLogProvider.NullProvider.class, BinaryLogProvider.load(cl).getClass());
   }
 
+  @Test
+  public void serverInterceptorFirstAndIsInputstream() throws Exception {
+    String serverName = getClass().getName() + "-serverInterceptor";
+    ServerInterceptor interceptor = new TracingServerInterceptor(req, resp, methods);
+    InProcessServerBuilder builder = InProcessServerBuilder
+        .forName(serverName)
```

Comment:
Add `directExecutor()`. This code is not thread-safe and it's simply easier to not think about it than proving it is correct.

Reply:
No longer applicable.